README.md
# SendState
SendState jest wtyczką do komunikatora [AQQ](http://www.aqq.eu/pl.php). Umożliwia wysyłanie indywidualnego statusu kontaktom z sieci XMPP na czas trwania sesji.

### Wymagania
Do skompilowania wtyczki SendState potrzebne jest:

* Embarcadero RAD Studio XE7 (C++Builder oraz Delphi, jeżeli chcemy kompilować wersję x64)
* Pełna wersja komponentów [AlphaControls](http://www.alphaskins.com/) (do kompilacji wersji x64 potrzebne jest Delphi)
* [Plik nagłówkowy SDK komunikatora AQQ](https://bitbucket.org/beherit/pluginapi-for-aqq-im)
* Opcjonalnie [UPX](http://upx.sourceforge.net/) dla zmniejszenia rozmiaru pliku wynikowego (w szczególności wersji x64)

### Błędy
Znalezione błędy wtyczki należy zgłaszać na [tracekerze](http://forum.aqq.eu/tracker/project-74-sendstate/) znajdującym na oficjalnym forum komunikatora AQQ lub pisząc bezpośrednio do autora wtyczki (preferowany kontakt poprzez Jabber).

### Kontakt z autorem
Autorem wtyczki SendState jest Krzysztof Grochocki. Możesz skontaktować się z nim drogą mailową (kontakt@beherit.pl) lub poprzez Jabber (im@beherit.pl).

### Licencja
Wtyczka SendState objęta jest licencją [GNU General Public License 3](http://www.gnu.org/copyleft/gpl.html).

~~~~
SendState
Copyright (C) 2010-2015  Krzysztof Grochocki

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
~~~~

CHANGELOG.md
1.4.0.0
-----
* Pełne wsparcie kolorystyki stylizacji okien.
* Obsługa lokalizacji (polskiej i angielskiej).
* Prawidłowe działanie z wieloma kontami XMPP.
* Inne kosmetyczne poprawki.
* Aktualizacja komponentów AlphaControls do nowszej wersji (9.20).

1.3.3.0
-----
* Poprawki w zapisywaniu w pamięci wysłanego opisu.
* Nieskórkowanie systemowych okien dialogowych niepochodzących z wtyczki.
* Aktualizacja komponentów AlphaControls do nowszej wersji (9.10).

1.3.2.0
-----
* Zablokowanie funkcjonalności dla kontaktów z sieci Facebook.
* Kompilacja pod poprawionymi nagłówkami dla wersji 64-bitowej.
* Zmniejszenie rozmiaru wtyczki 64-bitowej poprzez użycie kompresji LZMA.
* Aktualizacja komponentów AlphaControls do nowszej wersji (8.53).

1.3.1.0
-----
* Obsługa kolorystyki stylizacji okien (barwa i nasycenie).
* Optymalizacja kodu.
* Drobna poprawka w obsłudze nadawania przeźroczystości obramowań przez wtyczkę AlphaWindows.
* Aktualizacja komponentów AlphaControls do nowszej wersji (8.44).

1.3.0.0
-----
* Dodanie 64-bitowej wersji wtyczki.
* Wsparcie dla wtyczki AlphaWindows (ustawianie przeźroczystości obramowania).
* Zmniejszenie rozmiaru wtyczki 32-bitowej poprzez użycie kompresji LZMA.
* Drobne zmiany w skórkowaniu form.
* Aktualizacja komponentów AlphaControls do nowszej wersji (8.20).

1.2.1.0
-----
* Drobne poprawki w procesie wyładowania wtyczki.

1.2.0.0
-----
* Wysłany opis ustawiany jest w końcu na cały czas trwania sesji!
* Oznaczenie wysłania już statusu do kontaktu.
* Możliwość szybkiego przywrócenia aktualnego opisu.
* Przepisanie wtyczki pod kompilator Embarcadero C++ Builder XE3.
* Aktualizacja komponentów AlphaControls do najnowszej dostępnej wersji (7.70).
* Optymalizacja kodu.

1.1.2.0
-----
* Obsługa natywnego stylu Windows (wyłączone skórkowanie okien wtyczki).
* Aktualizacja komponentów AlphaControls do najnowszej dostępnej wersji (7.60).
* Inne mniej ważne poprawki.

1.1.0.0
-----
* Pełne skórkowanie okna wtyczki.
* Zapamiętywanie wysłanego wcześniej statusu do danego kontaktu.
* Optymalizacja kodu.

1.0.1.0
-----
* Poprawka w wysyłanym pakiecie XML.

1.0.0.2
-----
* Kodowanie niedozwolonych znaków w opisie.

1.0.0.0
-----
* Pierwsza publiczna wersja.

SendFrm.h
//---------------------------------------------------------------------------
// Copyright (C) 2010-2014 Krzysztof Grochocki
//
// This file is part of SendState
//
// SendState is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// SendState is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
#ifndef SendFrmH
#define SendFrmH
#define WM_ALPHAWINDOWS (WM_USER + 666)
//---------------------------------------------------------------------------
#include "acAlphaImageList.hpp"
#include "sButton.hpp"
#include "sComboBox.hpp"
#include "sMemo.hpp"
#include "sSkinManager.hpp"
#include "sSkinProvider.hpp"
#include "sSpeedButton.hpp"
#include <System.Actions.hpp>
#include <System.Classes.hpp>
#include <Vcl.ActnList.hpp>
#include <Vcl.Buttons.hpp>
#include <Vcl.Controls.hpp>
#include <Vcl.ImgList.hpp>
#include <Vcl.StdCtrls.hpp>
//---------------------------------------------------------------------------
class TSendForm : public TForm
{
__published:	// IDE-managed Components
	TsMemo *StatusMemo;
	TsButton *SendButton;
	TActionList *ActionList;
	TAction *aExit;
	TAction *aSelectMemo;
	TsSkinManager *sSkinManager;
	TsSkinProvider *sSkinProvider;
	TsAlphaImageList *sAlphaImageList;
	TsComboBox *StateComboBox;
	TsSpeedButton *ResetButton;
	void __fastcall aExitExecute(TObject *Sender);
	void __fastcall aSelectMemoExecute(TObject *Sender);
	void __fastcall SendButtonClick(TObject *Sender);
	void __fastcall FormShow(TObject *Sender);
	void __fastcall FormCreate(TObject *Sender);
	void __fastcall StateComboBoxDrawItem(TWinControl *Control, int Index, TRect &Rect,
          TOwnerDrawState State);
	void __fastcall ResetButtonClick(TObject *Sender);
	void __fastcall sSkinManagerSysDlgInit(TacSysDlgData DlgData, bool &AllowSkinning);
private:	// User declarations
public:		// User declarations
	UnicodeString JID;
	int UserIdx;
	__fastcall TSendForm(TComponent* Owner);
	void __fastcall WMTransparency(TMessage &Message);
	BEGIN_MESSAGE_MAP
	MESSAGE_HANDLER(WM_ALPHAWINDOWS,TMessage,WMTransparency);
	END_MESSAGE_MAP(TForm)
};
//---------------------------------------------------------------------------
extern PACKAGE TSendForm *SendForm;
//---------------------------------------------------------------------------
#endif


SendFrm.cpp
//---------------------------------------------------------------------------
// Copyright (C) 2010-2015 Krzysztof Grochocki
//
// This file is part of SendState
//
// SendState is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// SendState is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop
#include "SendFrm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "acAlphaImageList"
#pragma link "sButton"
#pragma link "sComboBox"
#pragma link "sMemo"
#pragma link "sSkinManager"
#pragma link "sSkinProvider"
#pragma link "sSpeedButton"
#pragma resource "*.dfm"
TSendForm *SendForm;
//---------------------------------------------------------------------------
__declspec(dllimport)UnicodeString GetThemeSkinDir();
__declspec(dllimport)bool ChkSkinEnabled();
__declspec(dllimport)bool ChkThemeAnimateWindows();
__declspec(dllimport)bool ChkThemeGlowing();
__declspec(dllimport)int GetHUE();
__declspec(dllimport)int GetSaturation();
__declspec(dllimport)int GetBrightness();
__declspec(dllimport)void SendXML(UnicodeString JID, int UserIdx, UnicodeString Status, int State);
__declspec(dllimport)UnicodeString GetIconPath(int Icon);
__declspec(dllimport)UnicodeString GetStatus(int UserIdx);
__declspec(dllimport)int GetState(int UserIdx);
//---------------------------------------------------------------------------
__fastcall TSendForm::TSendForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TSendForm::WMTransparency(TMessage &Message)
{
	Application->ProcessMessages();
	if(sSkinManager->Active) sSkinProvider->BorderForm->UpdateExBordersPos(true, (int)Message.LParam);
}
//---------------------------------------------------------------------------

void __fastcall TSendForm::aExitExecute(TObject *Sender)
{
	//Zamkniecie formy
	Close();
}
//---------------------------------------------------------------------------

void __fastcall TSendForm::FormCreate(TObject *Sender)
{
	//Wlaczona zaawansowana stylizacja okien
	if(ChkSkinEnabled())
	{
		UnicodeString ThemeSkinDir = GetThemeSkinDir();
		//Plik zaawansowanej stylizacji okien istnieje
		if(FileExists(ThemeSkinDir + "\\\\Skin.asz"))
		{
			//Dane pliku zaawansowanej stylizacji okien
			ThemeSkinDir = StringReplace(ThemeSkinDir, "\\\\", "\\", TReplaceFlags() << rfReplaceAll);
			sSkinManager->SkinDirectory = ThemeSkinDir;
			sSkinManager->SkinName = "Skin.asz";
			//Ustawianie animacji AlphaControls
			if(ChkThemeAnimateWindows()) sSkinManager->AnimEffects->FormShow->Time = 200;
			else sSkinManager->AnimEffects->FormShow->Time = 0;
			sSkinManager->Effects->AllowGlowing = ChkThemeGlowing();
			//Zmiana kolorystyki AlphaControls
			sSkinManager->HueOffset = GetHUE();
			sSkinManager->Saturation = GetSaturation();
			sSkinManager->Brightness = GetBrightness();
			//Aktywacja skorkowania AlphaControls
			sSkinManager->Active = true;
		}
		//Brak pliku zaawansowanej stylizacji okien
		else sSkinManager->Active = false;
	}
	//Zaawansowana stylizacja okien wylaczona
	else sSkinManager->Active = false;
}
//---------------------------------------------------------------------------

void __fastcall TSendForm::FormShow(TObject *Sender)
{
	//Wczytanie ikon z interfesju AQQ
	sAlphaImageList->AcBeginUpdate();
	sAlphaImageList->Clear();
	sAlphaImageList->LoadFromFile(GetIconPath(1));
	sAlphaImageList->LoadFromFile(GetIconPath(7));
	sAlphaImageList->LoadFromFile(GetIconPath(3));
	sAlphaImageList->LoadFromFile(GetIconPath(4));
	sAlphaImageList->LoadFromFile(GetIconPath(5));
	sAlphaImageList->LoadFromFile(GetIconPath(2));
	sAlphaImageList->LoadFromFile(GetIconPath(49));
	//Ustawienie ikonki na przycisku przywracania statusu
	ResetButton->Images = sAlphaImageList;
	ResetButton->ImageIndex = 6;
	//Ustawianie fokusa na polu wpisywania opisu
	StatusMemo->SetFocus();
	StatusMemo->SelectAll();
}
//---------------------------------------------------------------------------

void __fastcall TSendForm::aSelectMemoExecute(TObject *Sender)
{
	//Zaznaczenie calego tekstu
	StatusMemo->SelectAll();
}
//---------------------------------------------------------------------------

void __fastcall TSendForm::SendButtonClick(TObject *Sender)
{
	//Wyslanie nowego statusu
	SendXML(JID, UserIdx, StatusMemo->Text.Trim(), StateComboBox->ItemIndex);
	//Zamkniecie formy
	Close();
}
//---------------------------------------------------------------------------

void __fastcall TSendForm::StateComboBoxDrawItem(TWinControl *Control, int Index, TRect &Rect,
			TOwnerDrawState State)
{
	StateComboBox->Canvas->Brush->Style = bsClear;
	sAlphaImageList->Draw(StateComboBox->Canvas, Rect.Left+3, Rect.Top+3, Index);
	StateComboBox->Canvas->TextOutW(Rect.Left+22, Rect.Top+4, StateComboBox->Items->Strings[Index]);
}
//---------------------------------------------------------------------------

void __fastcall TSendForm::ResetButtonClick(TObject *Sender)
{
	//Przywracanie aktualnego opisu
	StatusMemo->Text = GetStatus(UserIdx);
	//Przywracanie aktualnego statusu
	StateComboBox->ItemIndex = GetState(UserIdx);
}
//---------------------------------------------------------------------------

void __fastcall TSendForm::sSkinManagerSysDlgInit(TacSysDlgData DlgData, bool &AllowSkinning)
{
	AllowSkinning = false;
}
//---------------------------------------------------------------------------


SendState.h
//---------------------------------------------------------------------------
// Copyright (C) 2010-2014 Krzysztof Grochocki
//
// This file is part of SendState
//
// SendState is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// SendState is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

#include <vcl.h>
#include <tchar.h>



SendStateFrm.h
//---------------------------------------------------------------------------
// Copyright (C) 2010-2014 Krzysztof Grochocki
//
// This file is part of SendState
//
// SendState is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// SendState is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
#ifndef SendStateFrmH
#define SendStateFrmH
#define WM_ALPHAWINDOWS (WM_USER + 666)
//---------------------------------------------------------------------------
#include "acAlphaImageList.hpp"
#include "sButton.hpp"
#include "sComboBox.hpp"
#include "sMemo.hpp"
#include "sSkinManager.hpp"
#include "sSkinProvider.hpp"
#include "sSpeedButton.hpp"
#include <System.Actions.hpp>
#include <System.Classes.hpp>
#include <Vcl.ActnList.hpp>
#include <Vcl.Buttons.hpp>
#include <Vcl.Controls.hpp>
#include <Vcl.ImgList.hpp>
#include <Vcl.StdCtrls.hpp>
//---------------------------------------------------------------------------
class TSendStateForm : public TForm
{
__published:	// IDE-managed Components
	TsMemo *StatusMemo;
	TsButton *SendButton;
	TActionList *ActionList;
	TAction *aExit;
	TAction *aSelectMemo;
	TsSkinManager *sSkinManager;
	TsSkinProvider *sSkinProvider;
	TsAlphaImageList *sAlphaImageList;
	TsComboBox *StateComboBox;
	TsSpeedButton *ResetButton;
	void __fastcall aExitExecute(TObject *Sender);
	void __fastcall aSelectMemoExecute(TObject *Sender);
	void __fastcall SendButtonClick(TObject *Sender);
	void __fastcall FormShow(TObject *Sender);
	void __fastcall FormCreate(TObject *Sender);
	void __fastcall StateComboBoxDrawItem(TWinControl *Control, int Index, TRect &Rect,
          TOwnerDrawState State);
	void __fastcall ResetButtonClick(TObject *Sender);
	void __fastcall sSkinManagerSysDlgInit(TacSysDlgData DlgData, bool &AllowSkinning);
private:	// User declarations
public:		// User declarations
	UnicodeString JID;
	int UserIdx;
	__fastcall TSendStateForm(TComponent* Owner);
	void __fastcall WMTransparency(TMessage &Message);
	BEGIN_MESSAGE_MAP
	MESSAGE_HANDLER(WM_ALPHAWINDOWS,TMessage,WMTransparency);
	END_MESSAGE_MAP(TForm)
};
//---------------------------------------------------------------------------
extern PACKAGE TSendStateForm *SendStateForm;
//---------------------------------------------------------------------------
#endif


SendStateFrm.cpp
//---------------------------------------------------------------------------
// Copyright (C) 2010-2015 Krzysztof Grochocki
//
// This file is part of SendState
//
// SendState is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// SendState is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with GNU Radio; see the file COPYING. If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
#include <vcl.h>
#include <LangAPI.hpp>
#pragma hdrstop
#include "SendStateFrm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "acAlphaImageList"
#pragma link "sButton"
#pragma link "sComboBox"
#pragma link "sMemo"
#pragma link "sSkinManager"
#pragma link "sSkinProvider"
#pragma link "sSpeedButton"
#pragma resource "*.dfm"
TSendStateForm *SendStateForm;
//---------------------------------------------------------------------------
__declspec(dllimport)UnicodeString GetThemeSkinDir();
__declspec(dllimport)bool ChkSkinEnabled();
__declspec(dllimport)bool ChkThemeAnimateWindows();
__declspec(dllimport)bool ChkThemeGlowing();
__declspec(dllimport)int GetHUE();
__declspec(dllimport)int GetSaturation();
__declspec(dllimport)int GetBrightness();
__declspec(dllimport)void SendXML(UnicodeString JID, int UserIdx, UnicodeString Status, int State);
__declspec(dllimport)UnicodeString GetIconPath(int Icon);
__declspec(dllimport)UnicodeString GetStatus(int UserIdx);
__declspec(dllimport)int GetState(int UserIdx);
//---------------------------------------------------------------------------
__fastcall TSendStateForm::TSendStateForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TSendStateForm::WMTransparency(TMessage &Message)
{
	Application->ProcessMessages();
	if(sSkinManager->Active) sSkinProvider->BorderForm->UpdateExBordersPos(true, (int)Message.LParam);
}
//---------------------------------------------------------------------------

void __fastcall TSendStateForm::aExitExecute(TObject *Sender)
{
	//Zamkniecie formy
	Close();
}
//---------------------------------------------------------------------------

void __fastcall TSendStateForm::FormCreate(TObject *Sender)
{
	//Lokalizowanie formy
	LangForm(this);
	//Wlaczona zaawansowana stylizacja okien
	if(ChkSkinEnabled())
	{
		UnicodeString ThemeSkinDir = GetThemeSkinDir();
		//Plik zaawansowanej stylizacji okien istnieje
		if(FileExists(ThemeSkinDir + "\\\\Skin.asz"))
		{
			//Dane pliku zaawansowanej stylizacji okien
			ThemeSkinDir = StringReplace(ThemeSkinDir, "\\\\", "\\", TReplaceFlags() << rfReplaceAll);
			sSkinManager->SkinDirectory = ThemeSkinDir;
			sSkinManager->SkinName = "Skin.asz";
			//Ustawianie animacji AlphaControls
			if(ChkThemeAnimateWindows()) sSkinManager->AnimEffects->FormShow->Time = 200;
			else sSkinManager->AnimEffects->FormShow->Time = 0;
			sSkinManager->Effects->AllowGlowing = ChkThemeGlowing();
			//Zmiana kolorystyki AlphaControls
			sSkinManager->HueOffset = GetHUE();
			sSkinManager->Saturation = GetSaturation();
			sSkinManager->Brightness = GetBrightness();
			//Aktywacja skorkowania AlphaControls
			sSkinManager->Active = true;
		}
		//Brak pliku zaawansowanej stylizacji okien
		else sSkinManager->Active = false;
	}
	//Zaawansowana stylizacja okien wylaczona
	else sSkinManager->Active = false;
}
//---------------------------------------------------------------------------

void __fastcall TSendStateForm::FormShow(TObject *Sender)
{
	//Wczytanie ikon z interfesju AQQ
	sAlphaImageList->AcBeginUpdate();
	sAlphaImageList->Clear();
	sAlphaImageList->LoadFromFile(GetIconPath(1));
	sAlphaImageList->LoadFromFile(GetIconPath(7));
	sAlphaImageList->LoadFromFile(GetIconPath(3));
	sAlphaImageList->LoadFromFile(GetIconPath(4));
	sAlphaImageList->LoadFromFile(GetIconPath(5));
	sAlphaImageList->LoadFromFile(GetIconPath(2));
	sAlphaImageList->LoadFromFile(GetIconPath(49));
	//Ustawienie ikonki na przycisku przywracania statusu
	ResetButton->Images = sAlphaImageList;
	ResetButton->ImageIndex = 6;
	//Ustawianie fokusa na polu wpisywania opisu
	StatusMemo->SetFocus();
	StatusMemo->SelectAll();
}
//---------------------------------------------------------------------------

void __fastcall TSendStateForm::aSelectMemoExecute(TObject *Sender)
{
	//Zaznaczenie calego tekstu
	StatusMemo->SelectAll();
}
//---------------------------------------------------------------------------

void __fastcall TSendStateForm::SendButtonClick(TObject *Sender)
{
	//Wyslanie nowego statusu
	SendXML(JID, UserIdx, StatusMemo->Text.Trim(), StateComboBox->ItemIndex);
	//Zamkniecie formy
	Close();
}
//---------------------------------------------------------------------------

void __fastcall TSendStateForm::StateComboBoxDrawItem(TWinControl *Control, int Index, TRect &Rect,
			TOwnerDrawState State)
{
	StateComboBox->Canvas->Brush->Style = bsClear;
	sAlphaImageList->Draw(StateComboBox->Canvas, Rect.Left+3, Rect.Top+3, Index);
	StateComboBox->Canvas->TextOutW(Rect.Left+22, Rect.Top+4, StateComboBox->Items->Strings[Index]);
}
//---------------------------------------------------------------------------

void __fastcall TSendStateForm::ResetButtonClick(TObject *Sender)
{
	//Przywracanie aktualnego opisu
	StatusMemo->Text = GetStatus(UserIdx);
	//Przywracanie aktualnego statusu
	StateComboBox->ItemIndex = GetState(UserIdx);
}
//---------------------------------------------------------------------------

void __fastcall TSendStateForm::sSkinManagerSysDlgInit(TacSysDlgData DlgData, bool &AllowSkinning)
{
	AllowSkinning = false;
}
//---------------------------------------------------------------------------
